subsys/bluetooth/audio/unicast_server.c

```diff
@@ -53,6 +53,36 @@ int bt_audio_unicast_server_unregister_cb(const struct bt_audio_unicast_server_c
 	return 0;
 }
 
+int bt_unicast_server_metadata(struct bt_audio_stream *stream,
+			       struct bt_codec_data meta[],
+			       size_t meta_count)
+{
+	struct bt_audio_ep *ep;
+	int err;
+
+
+	if (unicast_server_cb != NULL && unicast_server_cb->metadata != NULL) {
+		err = unicast_server_cb->metadata(stream, meta, meta_count);
+	} else {
+		err = -ENOTSUP;
+	}
+
+	ep = stream->ep;
+	for (size_t i = 0U; i < meta_count; i++) {
+		(void)memcpy(&ep->codec.meta[i], &meta[i],
+			     sizeof(ep->codec.meta[i]));
+	}
+
+	if (err) {
+		return err;
+	}
+
+	/* Set the state to the same state to trigger the notifications */
+	ascs_ep_set_state(ep, ep->status.state);
```

Comment:
Why would keeping the state trigger notifications?

Reply:
This is just copied code, but `ascs_ep_set_state` will call `ase_status_changed` to send notifications. The reason we do this is that setting metadata does not change the endpoint state, but the notification contains the metadata, so this is a way to use existing functionality to trigger the notification to send metadata. 

I could implement a `ascs_ep_metadata_changed` function to add some more readability to it if you like

Reply:
Thanks for the explanation.  Your proposal sounds good, but I do not have strong opinion on it.

Reply:
(Feel free to resolve any way.)

Reply:
I'd rather keep it as is, as changing this seems outside the scope of this PR (code was just moved). 

I still have a lot of other cleanup things in mind for BAP, so maybe it will it will automatically fixed as part of those :D